src/main/certificateManager.js

```diff
@@ -22,42 +22,20 @@ export class CertificateManager {
 
       // store callback so it can be called with selected certificate
       this.certificateRequestCallbackMap.set(url, callback);
-      this.addToCertificateQueue(url, list);
+      this.popCertificateModal(url, list);
     } else {
       log.info(`There were ${list.length} candidate certificates. Skipping certificate selection`);
     }
   }
 
-  addToCertificateQueue = (url, list) => {
-    this.certificateQueue.push({
-      url,
-      list
+  popCertificateModal = (url, list) => {
+    const modalPromise = addModal(`certificate-${url}`, html, modalPreload, {url, list}, WindowManager.getMainWindow());
+    modalPromise.then((data) => {
+      const {cert} = data;
+      this.handleSelectedCertificate(url, cert);
+    }).catch(() => {
```

Comment:
same as above, if something went wrong we could at least throw a line in the console stating what happened.

if we cound differentiate between user cancelling and app error that would be great :)